llvm/lib/IR/DataLayout.cpp

```diff
@@ -649,6 +649,8 @@ Error DataLayout::setPointerAlignmentInBits(uint32_t AddrSpace, Align ABIAlign,
   if (PrefAlign < ABIAlign)
     return reportError(
         "Preferred alignment cannot be less than the ABI alignment");
+  if (IndexBitWidth > TypeBitWidth)
```

Comment:
I think it would be good to add tests for this: You could use the test I added in https://reviews.llvm.org/D135158?

Reply:
There is a test here: https://github.com/llvm/llvm-project/pull/70015/files#diff-934d54edeeb3870404b0553b4964c06cde47e081976e180580f5ade6ee9bd2ab

Reply:
Somehow missed that while scrolling over the changed files. LGTM.